pkg/controllers/trace_controller.go

```diff
@@ -88,8 +121,21 @@ func (r *TraceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	// Lookup factory
 	factory, ok := r.TraceFactories[trace.Spec.Gadget]
 	if !ok {
-		log.Errorf("Unknown gadget %q", trace.Spec.Gadget)
-		return ctrl.Result{}, nil
+		return ctrl.Result{}, setTraceOpError(ctx, r.Client, req.NamespacedName.String(),
+			trace, fmt.Sprintf("Unknown gadget %q", trace.Spec.Gadget))
+	}
+
+	// Check trace specs before adding the finalizer and registering the trace
+	if trace.Spec.RunMode != "Manual" {
+		return ctrl.Result{}, setTraceOpError(ctx, r.Client, req.NamespacedName.String(),
+			trace, fmt.Sprintf("Unsupported RunMode %q for gadget %q",
+				trace.Spec.RunMode, trace.Spec.Gadget))
```

Comment:
If the Reconcile() function returns an error, the controller will call Reconcile() again and again with an exponential backoff, until it works. Is it what you want?

Reply:
No, and I would prefer to avoid letting the controller retying even if we would have the risk that the Status does not get updated. 

I will change it to not retry if there is an invalid parameter in the specs. I will update the PR to avoid this.